python/dgl/graphbolt/item_sampler.py

```diff
@@ -306,27 +325,147 @@ def _collate(batch):
         return default_collate(batch)
 
     def __iter__(self) -> Iterator:
-        data_pipe = IterableWrapper(self._item_set)
-        # Shuffle before batch.
-        if self._shuffle:
-            # `torchdata.datapipes.iter.Shuffler` works with stream too.
-            # To ensure randomness, make sure the buffer size is at least 10
-            # times the batch size.
-            buffer_size = max(10000, 10 * self._batch_size)
-            data_pipe = data_pipe.shuffle(buffer_size=buffer_size)
-
-        # Batch.
-        data_pipe = data_pipe.batch(
-            batch_size=self._batch_size,
-            drop_last=self._drop_last,
-        )
+        # Organize items.
+        data_pipe = self._organize_items(self._item_set)
 
         # Collate.
         data_pipe = data_pipe.collate(collate_fn=self._collate)
 
         # Map to minibatch.
-        data_pipe = data_pipe.map(
-            partial(self._minibatcher, names=self._item_set.names)
-        )
+        data_pipe = data_pipe.map(partial(self._minibatcher, names=self._names))
 
         return iter(data_pipe)
+
+
+class DistributedItemSampler(ItemSampler):
+    """Distributed Item Sampler.
+
+    This sampler creates a distributed subset of items from the given data set,
```

Comment:
I think we'd better clearly introduce the workflow: `shard -> shuffle -> ...` to help user have a clear understanding of this datapipe. And For this dist one, add note for the shuffle part which may lower the data diversity potentially.